source/documentation/deploying_services/route_services.md

```diff
@@ -0,0 +1,102 @@
+
+
+2. Bind this Route Service Instance to the [application route](/#names-routes-and-domains)
+
+   ``cf bind-route-service DOMAIN SERVICE_INSTANCE --hostname HOSTNAME``
+
+   where `DOMAIN` is the domain used by your application, likely the default one `cloudapps.digital`, `SERVICE_INSTANCE` the name of the service that you have just created, and `HOSTNAME` the host or app name assigned to the app. For example:
+
+   ``cf bind-route-service cloudapps.digital my-route-service --hostname myapp``
+
+3. You can list the [routes](/#names-routes-and-domains) of the current [space](https://docs.cloud.service.gov.uk/#organisations-spaces-amp-targets), to learn the applications and Route Services bound to them:
+
+   ``cf routes``
+
+If the Route Service endpoint is not responding correctly, you might get the following response when querying the route:
+
+   ``502 Bad Gateway: Registered endpoint failed to handle the request.``
+
+If you get this error, double check that the endpoint is working and reachable from the platform, that it is using a valid SSL certificate, that responds timely and that it implements the [Route Service Protocol](/#implementing-a-route-service).
+
+### Implementing a Route Service
+
+A Route Service can be implemented as a HTTPS application that will receive all requests for the associated route from the Cloud Foundry router. The request will contain the following additional headers:
+
+- `X-CF-Forwarded-Url` header contains the URL of the application route. The route service should forward the request to this URL.
+- `X-CF-Proxy-Signature` and `X-CF-Proxy-Metadata`: Signature and metadata used by the platform route itself to validate the request sent by the route service.
+  The route service must always include these headers.
+
+The Route Services must proxy back the request to the application route defined in header `X-CF-Forwarded-Url` within 60 seconds, otherwise the request will be refused. In addition, the total time to process the request and send a response back must be within 900 seconds.
+
+![Route Service request life cycle](images/route-service.png)
+
+You can refer to the [Cloud Foundry documentation](https://docs.cloudfoundry.org/services/route-services.html#service-instance-responsibilities), [example route services](https://docs.cloudfoundry.org/services/route-services.html#examples), and [tutorial](https://docs.cloudfoundry.org/services/route-services.html#tutorial) to learn more about how to implement route services.
+
+### Example: Route Service to add authentication
+
+In the following example we will add a Route Service to provide basic HTTP authentication to the bound routes. This can be used to restrict access to a beta application.
+
+The Route Service code can be found in https://github.com/alext/cf_basic_auth_route_service and we will deploy it as an application in the platform it self. We will bind this Route Service to a deployed application accesible via https://myapp.cloudapps.digital.
```

Comment:
The way this is phrased suggests that this is an application we're recommending that tenants deploy. I don't think that's the case, this is merely a proof-of-concept implementation that I knocked up. It's probably fine to point at as an example of what an implementation could look like, but not as an application that's implicitly supported etc.